Feature/3

solmoonkang opened the pull request:
1. fix : ResponseFormat 구현
2. fix : Exception 처리를 위한 Handler 구현
3. test 완료

solmoonkang committed:
savePost API 구현

solmoonkang committed:
boardInfo API 생성

solmoonkang committed:
BoardRepository 및 PostSaveDto 생성

solmoonkang committed:
findAllboardInfo API 구현

solmoonkang committed:
editPost API 구현

solmoonkang committed:
deletePost API 구현

solmoonkang committed:
findBoardInfo API 구현

solmoonkang committed:
fix : Users API

solmoonkang committed:
fix : UserController, Service, Repository name fix

solmoonkang committed:
fix : UserController, Service API fix

solmoonkang committed:
Post title, writer, content 미입력 시 예외처리 구현

solmoonkang committed:
postSave Exception 구현

solmoonkang committed:
fix : Board, PostSaveDto modifiedDate 삭제

solmoonkang committed:
editPost Exception 구현

solmoonkang committed:
deletePost Exception 구현

solmoonkang committed:
fix : UserController, UsersInfoDto fix

solmoonkang committed:
fix : findPostInfo명으로 fix, BoardController fix

solmoonkang committed:
fix : EditPostDto, PostInfoDto location fix

solmoonkang committed:
fix : findByBoardId(boardId) delete

solmoonkang committed:
fix : findAllPostInfo URL fix

solmoonkang committed:
fix : findAllPostInfo로 수정

solmoonkang committed:
fix : deletePost API fix

solmoonkang committed:
fix : findOnePostInfo API fix

solmoonkang committed:
search API paging 구현

solmoonkang committed:
fix : search API paging fix

solmoonkang committed:
findAllPostInfo API 구현
fix : search API fix
DeletePostDto, PostInfoDto delete

solmoonkang committed:
fix

solmoonkang committed:
fix : package structure change

solmoonkang committed:
fix : package structure change, class name change

solmoonkang committed:
test

solmoonkang committed:
DB 연동

solmoonkang committed:
DB 연동

solmoonkang committed:
test

solmoonkang committed:
fix: resolve

solmoonkang committed:
fix: DB 연동 및 Entity 수정

solmoonkang committed:
fix: 기존 Exception 삭제 및 User Exception 구현

solmoonkang committed:
fix: resolve

solmoonkang committed:
fix: Board Exception 수정

solmoonkang committed:
Valid 처리

solmoonkang committed:
fix : findAllUserInfo API 수정

solmoonkang committed:
fix : user exception 수정

solmoonkang committed:
fix : user exception 수정

solmoonkang committed:
fix

solmoonkang committed:
board validation 처리

hongdosan commented on src/main/java/kr/co/itresumeregistersite/controller/board/BoardController.java:
```diff
@@ -0,0 +1,54 @@
+public class BoardController {
+
+    private final BoardService boardService;
+
```
컨트롤러에서 응답값 ResponseFormat 만들어서 응답해주세요

hongdosan replied:
모든 컨트롤러에 해당하는 말입니다.

hongdosan replied:
또한 예외처리를 위해서 핸들러 처리도 해주세요.

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/entity/resume/Resume.java:
```diff
@@ -18,7 +19,7 @@ public class Resume {
 
     @ManyToOne
     @JoinColumn(name = "User_Id", referencedColumnName = "ID")
```
referencedColumnName = "ID" 이 속성값 지우세요.

hongdosan commented on src/main/java/kr/co/itresumeregistersite/service/board/BoardService.java:
```diff
@@ -0,0 +1,96 @@
+    public void postSave(SavePostDto savePostDto) {
+
+        // 제목, 작성자, 내용 미입력 시 예외 발생
+        NoInputTitle(savePostDto.getTitle());
```
왜 메소드 첫 문자가 대문자죠?

solmoonkang replied:
board 관련 기능들은 지금 수정 중이고 수정한거 다시 피알 올리겠습니당

HyuckJuneHong commented:
테스트도 진행해주세요 ㅎ

solmoonkang committed:
fix : exception 수정

solmoonkang committed:
ResponseFormat 구현

HyuckJuneHong commented:
아 글구 피알 올릴때 위에 설명도 꼭 써주세요 중요한 거나 꼭 봤으면 하는 것들 혹은 내가 뭘했는지 리스트들 작성해주셔야 제가 리뷰할 때 잘 해드릴 수 있어요

solmoonkang committed:
fix : ResponseFormat 수정, Exception명 수정, ExceptionAdvice 수정

solmoonkang committed:
Structure Change

solmoonkang commented:
네 알겠습니당

solmoonkang committed:
fix : savePost API, findAllPostInfo API fix

solmoonkang committed:
fix : comment

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/board/controller/BoardController.java:
```diff
@@ -0,0 +1,56 @@
+
+    // 게시글 전체 목록 조회
+    @GetMapping("/findAll")
+    public List<PostInfoDto> findAllPostInfo() {
```
List(PostInfoDto)도 ResponseFormat에 담아서 보낼 수 있어요~

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/board/repository/BoardRepository.java:
```diff
@@ -0,0 +1,19 @@
+@Repository
+public interface BoardRepository extends JpaRepository<Board, Long> {
+
+    Board findByBoardId(Long boardId);
```
Optional?

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/board/service/BoardService.java:
```diff
@@ -0,0 +1,99 @@
+
+
+    // 제목 작성 여부 검사
+    public void noInputTitle(String title) {
```
해당 메소드가 해당 패키지에서만 쓰이는 메소드라면 private로 선언하는게 좋을 것 같네여/

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/board/entity/Board.java:
```diff
@@ -0,0 +1,55 @@
+@Table(name = "tbl_board")
+public class Board {
+
+    @Id @GeneratedValue(strategy = GenerationType.AUTO)
```
GenerationType을 AUTO로 한 이유가 있나요?

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/user/controller/UserController.java:
```diff
@@ -0,0 +1,66 @@
+
+    // 전체 회원정보 조회
+    @GetMapping("/findAll")
+    public List<UserInfoDto> findAllUserInfo() {
```
ResponseFormat

HyuckJuneHong commented:
test?

solmoonkang committed:
fix :
1. findAllPostInfo ResponseFormat 추가
2. search ResponseFormat 추가
3. findAllPostInfo, search, editPost, deletePost API fix
4. Exception 추가
5. test 실행 : 성공

solmoonkang committed:
fix

solmoonkang committed:
fix

solmoonkang committed:
fix : editPost API modifiedDate 적용

hongdosan commented on src/main/java/kr/co/itresumeregistersite/domain/board/entity/Board.java:
```diff
@@ -0,0 +1,57 @@
+                     LocalDateTime modifiedDate) {
+        this.title = title;
+        this.content = content;
+        this.modifiedDate = LocalDateTime.now();
```
시간 값을 자동으로 넣어주는 Auditing 기능이 있는데 추후에 한번 공부하고 적용해보는 것도 좋을 것 같네요.

solmoonkang replied:
녜

hongdosan reviewed (approved):
아주 좋네용~
